mtags/src/main/scala-3/scala/meta/internal/pc/AutoImports.scala

```diff
@@ -160,77 +160,118 @@ object AutoImports extends AutoImportsBackticks:
+    def inferSymbolImport(symbol: Symbol): SymbolImport =
       indexedContext.lookupSym(symbol) match
-        case IndexedContext.Result.Missing => Some(AutoImport.Simple(symbol))
+        case IndexedContext.Result.Missing =>
+          // in java enum and enum case both have same flags
+          val enumOwner = symbol.owner.companion
+          def isJavaEnumCase: Boolean =
+            symbol.isAllOf(EnumVal) && enumOwner.is(Enum)
+
+          val (name, sel) =
+            // For enums import owner instead of all members
+            if symbol.isAllOf(EnumCase) || isJavaEnumCase
+            then
+              val ownerImport = inferSymbolImport(enumOwner)
+              (
+                SymbolIdent.Select(
+                  ownerImport.ident,
+                  symbol.nameBacktickedImport,
+                ),
+                ownerImport.importSel,
+              )
+            else
+              (
+                SymbolIdent.direct(symbol.nameBackticked),
+                Some(ImportSel.Direct(symbol)),
+              )
+          end val
+
+          SymbolImport(
+            symbol,
+            name,
+            sel,
+          )
         case IndexedContext.Result.Conflict =>
           val owner = symbol.owner
           renames(owner) match
             case Some(rename) =>
-              Some(AutoImport.renamedOrSpecified(symbol, rename))
-            case _ => None
-        case IndexedContext.Result.InScope => None
-
-    private def specifyOwnerEdit(symbol: Symbol, owner: String): l.TextEdit =
-      val line = pos.startLine
-      new l.TextEdit(pos.toLsp, s"$owner.${symbol.nameBacktickedImport}")
-
-    private def importEdit(
-        values: List[AutoImport],
-        importPosition: AutoImportPosition,
-    )(using Context): l.TextEdit =
-      val indent = " " * importPosition.indent
-      val topPadding =
-        if importPosition.padTop then "\n"
-        else ""
-
-      val formatted = values
-        .map({
-          case AutoImport.Simple(sym) => importName(sym)
-          case AutoImport.SpecifiedOwner(sym) => importName(sym.owner)
-          case AutoImport.Renamed(sym, rename) =>
-            s"${importName(sym.owner.owner)}.{${sym.owner.nameBacktickedImport} => $rename}"
-          case AutoImport.SelfRenamed(sym, rename) =>
-            s"${importName(sym.owner)}.{${sym.nameBacktickedImport} => $rename}"
-        })
-        .map(selector => s"${indent}import $selector")
-        .mkString(topPadding, "\n", "\n")
-
-      val editPos = pos.withSpan(Spans.Span(importPosition.offset)).toLsp
-      new l.TextEdit(editPos, formatted)
-    end importEdit
+              val importSel =
+                if rename != owner.showName then
+                  Some(ImportSel.Rename(owner, rename)).filter(_ =>
+                    !indexedContext.hasRename(owner, rename)
+                  )
+                else
+                  Some(ImportSel.Direct(owner)).filter(_ =>
```

Comment:
What if owner also have conflict, shouldn't we go up the onwer chain in this case?

Reply:
> What if owner also have conflict, shouldn't we go up the onwer chain in this case?

Yep, that's what I noticed too but when I started doing that properly it opened much more questions than I expected and it led to a bigger rework. At least right now it works as previously + as in Scala2

I have wip branch for that.

Reply:
Ok sure, we can improve it later on. If you don't manage to create a PR, could you open an issue mentioning that?

Reply:
I'm planning to turn it into PR during this week.
If it won't happen I will create an issue